crates/linker/src/pe.rs

```diff
@@ -183,17 +183,26 @@ pub(crate) fn preprocess_windows(
     host_exe_filename: &Path,
     metadata_filename: &Path,
     preprocessed_filename: &Path,
-    dummy_dll_symbols: &[String],
+    shared_lib: &Path,
     _verbose: bool,
     _time: bool,
 ) -> object::read::Result<()> {
-    let data = open_mmap(host_exe_filename);
+    let exec_data = open_mmap(host_exe_filename);
+    let shared_lib_data = &*open_mmap(shared_lib);
+    let shared_lib_obj = match object::File::parse(shared_lib_data) {
+        Ok(obj) => obj,
+        Err(e) => internal_error!("Failed to parse shared library file: {e}"),
+    };
+    let dummy_dll_symbols = shared_lib_obj
+        .symbols()
```

Comment:
This may just _need_ `.dynamic_symbols()`. I am erring on the side of over-counting here so that at worse we allocate too much space in the resulting mmap.